packages/cardano-graphql-services/src/StakePoolSearch/StakePoolSearchHttpServer.ts

```diff
@@ -0,0 +1,71 @@
+import { Cardano } from '@cardano-sdk/core';
+import { DbSyncStakePoolSearchProvider } from './DbSyncStakePoolSearchProvider';
+import { ErrorObject, serializeError } from 'serialize-error';
+import { HttpServer } from '../Http';
+import { Logger, dummyLogger } from 'ts-log';
+import bodyParser, { Options } from 'body-parser';
+import express, { Router } from 'express';
+import net from 'net';
+
+export interface StakePoolSearchServerDependencies {
+  stakePoolSearchProvider: DbSyncStakePoolSearchProvider;
+  logger?: Logger;
+}
+
+export interface StakePoolSearchServerConfig {
+  listen: net.ListenOptions;
+  bodyParser?: {
+    limit?: Options['limit'];
+  };
+}
+
+export interface StakePoolSearchResponse {
+  stakePools: Cardano.StakePool[];
+}
+
+export class StakePoolSearchHttpServer extends HttpServer {
+  private constructor(
+    config: StakePoolSearchServerConfig,
+    router: Router,
+    dependencies: StakePoolSearchServerDependencies
+  ) {
+    super({ listen: config.listen, name: 'StakePoolSearchServer', router }, dependencies.logger);
+  }
+  static create(
+    config: StakePoolSearchServerConfig,
+    { stakePoolSearchProvider, logger = dummyLogger }: StakePoolSearchServerDependencies
+  ) {
+    const router = express.Router();
+    router.use(bodyParser.raw({ limit: config.bodyParser?.limit || '500kB', type: 'application/json' }));
+    router.get('/health', async (req, res) => {
+      logger.debug('/health', { ip: req.ip });
+      const body = { ok: true };
+      res.send(body);
+    });
+    router.post('/search', async (req, res) => {
```

Comment:
I'm wondering if we should use provider method names for paths in url implementations, so this would be `/querystakepools`. Thoughts @rhyslbw ?

Reply:
Agree. Let's implement the existing names, but review to see if we want to commit to an API with those names. I certainly think `querystakepools` isn't great.

Reply:
Sounds good. As for committing to names, it doesn't have to be hardcoded. Http client class could have a prop to map methods to paths, e.g.
```
{
  paths: {
    [keyof StakePoolSearchProvider]: string;
  }
}
```

Reply:
Right on, that's what we want for sure.